File: src/api/account.ts
import { api } from '@/lib/axios'
import type { User, UpdateAccountRequest } from '@/types'

type RawUser = {
  user_id?: number
  id?: number
  user_name?: string | null
  user_mail?: string | null
  user_picture?: string | null
}

const mapUser = (user: RawUser): User => ({
  user_id: user.user_id ?? user.id ?? 0,
  user_name: user.user_name ?? '',
  user_mail: user.user_mail ?? '',
  user_picture: user.user_picture ?? '',
})

export const accountApi = {
  /** GET /api/Account/account/:id */
  getById: async (id: number): Promise<User> => {
    const res = await api.get<RawUser>(`/Account/account/${id}`)
    return mapUser(res.data)
  },

  /** POST /api/Account/account — met à jour nom, email, avatar */
  update: async (data: UpdateAccountRequest): Promise<void> => {
    await api.post('/Account/account', data)
  },

  /** POST /api/Account/account-password — met à jour le mot de passe */
  updatePassword: async (data: UpdateAccountRequest): Promise<void> => {
    await api.post('/Account/account-password', data)
  },

  /** GET /api/User/all */
  getAllUsers: async (): Promise<User[]> => {
    const res = await api.get<RawUser[]>('/User/all')
    return res.data.map(mapUser)
  },
}

File: csharp/Models/Account/Account.cs
namespace Thiskord_Back.Models.Account
{
    public class UserAccount
    {
        public int user_id {  get; set; }
        public string user_name { get; set; }
        public string user_password { get; set; }

        public string user_mail { get; set; }

        public string user_picture { get; set; }
        public UserAccount() { }

        public UserAccount(int _user_id, string _user_name, string _user_password, string _user_mail, string _user_picture)
        {
            this.user_id = _user_id;
            this.user_name = _user_name;
            this.user_password = _user_password;
            this.user_mail = _user_mail;
            this.user_picture = _user_picture;
        }

        public UserAccount(int _user_id, string _user_name, string _user_mail, string _user_picture)
        {
            this.user_id = _user_id;
            this.user_name = _user_name;
            this.user_mail = _user_mail;
            this.user_picture = _user_picture;
        }
        public UserAccount(string _user_name, string _user_mail, string _user_password, string _user_picture)
        {
            this.user_name = _user_name;
            this.user_mail = _user_mail;
            this.user_password = _user_password;
            this.user_picture = _user_picture;
        }
    }
}
